Update Dedoc Verison and Spatie Query Builder Version.

udawadi007 opened the pull request:
Fixed issues with sort-names.
Update dependency version.

layerok committed:
Update README.md

layerok committed:
update README.md

layerok committed:
Update composer.json

layerok committed:
Update README.md

layerok committed:
Update README.md

layerok committed:
Update README.md

layerok committed:
Update README.md

layerok committed:
Update README.md

layerok committed:
Update README.md

layerok committed:
Update README.md

layerok committed:
change naming

layerok committed:
Update composer.json

layerok committed:
Update composer.json

layerok committed:
update namespace

layerok committed:
update README.md

layerok committed:
update licence

layerok committed:
refactor

exonn-ozies committed:
Updated scramble dependency

exonn-ozies committed:
Merge pull request #1 from exonn-GmbH/l11-compatibility

Laravel 11 Compatibility

layerok committed:
support getting values from controller method or controller property

This is now supported
->allowedIncludes($this->includes)
->allowedIncludes($this->includes())

layerok committed:
fix: feature wasn't documented if previous one wasn't detected

layerok committed:
sort query param as enum

layerok committed:
massive rework

udawadi007 committed:
Update to latest version of scramble and query-builder.

udawadi007 committed:
Update to latest version of scramble and query-builder.

udawadi007 committed:
Update to latest version of scramble and query-builder.

udawadi007 committed:
Update to latest version of scramble and query-builder.

udawadi007 committed:
Update to latest version of scramble and query-builder.

udawadi007 committed:
refactor: Update AllowedSortsExtension to use ObjectType for query parameter schema

udawadi007 committed:
Added missing object type on short

udawadi007 committed:
Added code to support custom Allowed sort.

udawadi007 committed:
update dedoc scramble version

layerok committed:
add basic tests

layerok committed:
fix test

layerok commented:
Hi, thanks for contribution.
I noticed you did some changes that will break AllowedFilterModesExtension. I added tests that you can run to be sure that your changes didn't break anything.

udawadi007 committed:
Merge remote-tracking branch 'upstream/main'

udawadi007 committed:
Spatie Query builder does not include parameter "filter_mode" so we are going to exception handle with default value for user who are using Latest stable version of:
"dedoc/scramble": "^0.11.0",
"spatie/laravel-query-builder": "^6.0"

Update sorts extension with available sort parameter.

layerok commented on src/AllowedFilterModesExtension.php:
```diff
@@ -30,7 +30,7 @@ public function handle(Operation $operation, RouteInfo $routeInfo)
         $values = $helper->inferValues($methodCall, $routeInfo);
 
-        $parameter = new Parameter(config(AllowedFilter::FilterModesQueryParamConfigKey), 'query');
+        $parameter = new Parameter(config(AllowedFilter::FilterModesQueryParamConfigKey)??"filter_mode", 'query');
```
you can just exclude `AllowedFilterModesExtension` from scramble extensions (config/scramble.php). 
That way you won't get any exceptions for reading non-existent config key. There is no need to add this change. Please, remove it

udawadi007 replied:
Yep. I was missing that. Will update on that.  Thank you.

layerok commented:
here is command to run tests
```php
./vendor/bin/pest
```

layerok committed:
add command for running tests

layerok committed:
Update README.md

udawadi007 committed:
Update Sort Description, Query builder except sort fields to be string.

Even though Test will be failed, as it does not match with existing expected value and structure. Also current sort doesn't show which key are available as sort option. So include those information on description, with-out affection other flows of sort. Also work well with Dedoc Scramble Documentation testing.

Revert back to original author for Allowed FilterModes Extension.

udawadi007 committed:
Added scope filter parameter.

layerok committed:
update scramble and laravel-query-builder version

layerok committed:
support multiple versions of scramble

udawadi007 committed:
Merge branch 'main' into main

udawadi007 committed:
Update filter

udawadi007 committed:
Update filter to read scope filter with default.

udawadi007 committed:
Update composer.json to support dedoc/scramble version 0.12.0

udawadi007 committed:
feat: add support for dedoc/scramble ^0.13.0
